### ISSUE TITLE
Verify and add test case for flatten

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/clinical_quality_language#627]
The following flatten expression should translate correctly:

```
define "Flatten Lists and Elements":
  flatten { { 1, 2, 3 }, 2, 3, 4 }
```

Verify that it does and add it as a test case to the translator regression suite.

**Comment by cmoesel** [comment]
When I tested this yesterday w/ the 1.5.2 engine, it _looked_ like it translated OK to me (wrapped the single numbers in `ToList` calls).

**Comment by brynrhodes** [comment]
This test works fine if list promotion is allowed, but if list promotion is disallowed (which is the default for quality improvement artifacts), the list inference should be List<Any>, but it incorrectly allows the Any to be downcast to an Integer because Any is "compatible" with any other type. Once a list element type has been inferred as Any, it needs to remain there. Further, because the signature of Flatten requires List<List<T>>, the resolution fails for List<Any> and the operator cannot be resolved. This is inconsistent with the intent of allowing flatten to work with Lists of mixed elements of Lists and Singletons. I'll log a separate issue to address this.

**Comment by brynrhodes** [comment]
Included in 1.5.5 Release
